agent/consul/server_test.go

```diff
@@ -1284,7 +1273,11 @@ func TestServer_Expect_NonVoters(t *testing.T) {
 	}
 
 	t.Parallel()
-	dir1, s1 := testServerDCExpectNonVoter(t, "dc1", 2)
+	dir1, s1 := testServerWithConfig(t, func(c *Config) {
+		c.Bootstrap = false
+		c.BootstrapExpect = 2
+		c.ReadReplica = true
+	})
```

Comment:
This was the only caller of `testServerDCExpectNonVoter` so I moved the settings here instead of deprecating the function.

The other ones have a few dozen callers, so I left them.